app/Livewire/TrialBalance/AddTrialBalance.php

```diff
@@ -147,9 +151,94 @@ public function update()
+        }
+        $filePath = 'public/uploads/' . $tbExportFormat . '.xlsx';
+        $newFilePath = 'uploads/' . $tbExportFormat . '.xlsx';
+        Storage::copy($filePath, $newFilePath);
+        $spreadsheet = IOFactory::load(storage_path('app/' . $newFilePath));
+
+        // load export config
+        $tbExportConfig = DB::select('SELECT template FROM report_templates WHERE template_name = ?', [$tbExportFormat]);
+        $jsonConfig = array_column($tbExportConfig, 'template')[0];
+        $jsonConfig = json_decode($jsonConfig, true);
+
+        // query data from GL
+        $queryMonth = date('m', strtotime($this->tbDate));
+        $queryYear = date('Y', strtotime($this->tbDate));
+        $this->interimPeriod = trim($this->interimPeriod);
+        if ($this->interimPeriod == 'Quarterly') {
+            $months = [];
+            switch ($this->quarter) {
+                case 'Q1':
+                    $months = ['01', '02', '03'];
+                    break;
+                case 'Q2':
+                    $months = ['04', '05', '06'];
+                    break;
+                case 'Q3':
+                    $months = ['07', '08', '09'];
+                    break;
+                case 'Q4':
+                    $months = ['10', '11', '12'];
+                    break;
+            }
+            $res = DB::select("SELECT ls_account_title_code,ls_total_credit,ls_total_debit FROM ledgersheet_total_debit_credit WHERE ls_summary_month IN ('" . implode("','", $months) . "') AND ls_summary_year = '$queryYear'");
+        } else if ($this->interimPeriod == 'Monthly') {
+            $res = DB::select("SELECT ls_account_title_code,ls_total_credit,ls_total_debit FROM ledgersheet_total_debit_credit WHERE ls_summary_month LIKE '$queryMonth%' AND ls_summary_year = '$queryYear'");
+        } else {
+            $res = DB::select("SELECT ls_account_title_code,ls_total_credit,ls_total_debit FROM ledgersheet_total_debit_credit WHERE ls_summary_year = '$queryYear'");
+        }
+
+        // queried data
+        $credits = array_column($res, 'ls_total_credit');
+        $debits = array_column($res, 'ls_total_debit');
+        $accountCodes = array_column($res, 'ls_account_title_code');
+        // GL accounts codes is formatted as: `<code> - <title>`
+        // we only need the <code>
+        for ($index = 0; $index < count($accountCodes); $index++) {
+            $accountCodes[$index] = trim(explode('-', $accountCodes[$index])[0]);
+        }
+
+        // key : val == rowNumber : fsData
+        $exportConfig = [];
+        $i = 0;
+        foreach ($jsonConfig as $key => $value) {
+            if (array_key_exists($accountCodes[$i], $jsonConfig)) {
+                $exportConfig[$value] = [
+                    'credit' => $credits[$i],
+                    'debit' => $debits[$i],
+                ];
+            }
+            $i += 1;
+            if ($i >= count($accountCodes)) {
+                break;
+            }
+        }
```

Comment:
Additional, does this handle quarterly scenarios? I think may repetition ng account codes if quarterly and we need to get the sum of the 3 months

Reply:
no it does not handle that, so does annual. ill fix that later

Reply:
@am-cid Sa annual I think okay lang since summary na yun ng buong year, no need to calculate. Yung quarterly lang talaga

Add ko lang din, kindly handle if walang entries na nakuha/account code for export config for cases like this (1st empty array is fetched data from GL, 2nd our jsonconfig):
![image](https://github.com/chaaals/se-financial-reporting/assets/70326902/faed6760-70ec-49de-9a1d-2b31ea8c216e)